internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,255 @@
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+type rule = aggregationv1.KeyerConfiguration_Fragment_Rule
+type matchPredicate = aggregationv1.MatchPredicate
+type repeatedResultPredicate = aggregationv1.ResultPredicate_RepeatedResultPredicate
+
+// Mapper defines the interface that Maps an incoming request to an aggregation key
+type Mapper interface {
```

Comment:
It may be worth moving the interface to the external `/pkg` directory so that others can define their own mapper implementation if they choose to.

Reply:
yes i considered that, but did not take the approach in order to be opinionated about rule parsing in xds-relay. Would we not improve the rules rather than exposing the interface on feature asks?